backend/src/middleware/validation.ts

```diff
@@ -30,11 +41,51 @@ const validateRequest: RequestHandler = (req, res, next) => {
   throw createHttpError(400, errorString);
 };
 
+const validateUser: RequestHandler = asyncHandler(async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  let token;
+
+  if (authHeader && authHeader.split(" ")[0] === "Bearer") {
+    token = authHeader.split(" ")[1];
+  } else {
+    throw createHttpError(401, "Authorization Token Required");
+  }
+
+  try {
+    const decodedToken = await firebaseAdmin.firebaseAuth.verifyIdToken(token);
+    const email = decodedToken.email;
+
+    const user = await UserModel.findOne({ email });
```

Comment:
Nitpick (obviously doesn't really matter, just something to think about): Might be nicer to extract fetching a single user by email into a function in `backend/src/services/user.ts`